crates/pyhq/src/client/job.rs

```diff
@@ -25,6 +26,7 @@ use crate::{borrow_mut, run_future, ClientContextPtr, FromPyObject, PyJobId, PyT
 pub struct ResourceRequestDescription {
     n_nodes: NumOfNodes,
     cpus: String,
+    generic: Option<HashMap<String, u64>>,
```

Comment:
Could this be just a dictionary, without the `Option`? It seems unnecessary. We can use `dataclass.field(default_factory=dict)` for initializing the `generic` field.